File: migration-sql-firestore/src/main/java/com/otakuma/migration/model/Categorie.java
package com.otakuma.migration.model;

import lombok.Data;
import lombok.ToString;

@Data
public class Categorie {
    private Long categorieID;
    private String code;
    private String nom;
    private Integer nombreProduits;
    private Integer qte;
    private Integer pendingQte;
    private Integer activeQte;
    private Long categorieParent;
    private String keywords;
    private String description;
    private Boolean isActive;
    private String smallImage;
    private String mediumImage;
    private String largeImage;
    private String extra1;
    private String extra2;
    private String extra3;
}

File: otafront/src/app/data/Categorie.ts
import { Component, OnInit, Input } from '@angular/core';

export class Categorie {

    public code: string;
    public keywords: string;
    public extra1: string;
    public extra2: string;
    public extra3: string;
    
    public link: string;
    
    @Input() public nom: string ;
    @Input() public description: string;
    @Input() public isActive: boolean;
    @Input() public smallImage: string;
    @Input() public mediumImage: string;
    @Input() public largeImage: string;
    
    public nombreProduits: number;
    public qte: number; 
    public activeQte: number;
    public pendingQte: number;

    constructor() {
    }
}
